metricbeat/module/kubernetes/state_horizontalpodautoscaler/state_horizontalpodautoscaler.go

```diff
@@ -0,0 +1,47 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package state_horizontalpodautoscaler
+
+import (
+	"github.com/elastic/beats/v7/metricbeat/helper/kubernetes"
+	p "github.com/elastic/beats/v7/metricbeat/helper/prometheus"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/metricbeat/module/kubernetes/util"
+)
+
+// mapping stores the state metrics we want to fetch and will be used by this metricset
+var mapping = &p.MetricsMapping{
+	Metrics: map[string]p.MetricMap{
```

Comment:
From the list here I see that:  kube_horizontalpodautoscaler_spec_target_metric (although is experimental) that might also be interesting

WDYT? @MichaelKatsoulis

Reply:
I am not 100% sure what this metric mean. But why not add it? Maybe @endorama whose team actually uses this metrics could argue about its importance.

Reply:
This metric represent the target metric utilization that HPA uses in calculation. In our example the HPA was set to trigger on 30% CPU utilization:
```yaml
  metrics:
  - type: Resource
    resource:
      name: cpu
      target:
        type: Utilization
        averageUtilization: 30
```

Definitely a useful metrics to understand why a HPA behaves in a certain way but I would not considered it critical and we have not planned to use it.
I'm also not familiar with policies around including experimental metrics and I would favor keeping the maintenance cost low unless we have a use case that requires it.